src/main/java/org/opensearch/ad/ProfileUtil.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.ad;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.ad.constant.CommonName;
+import org.opensearch.ad.model.AnomalyDetector;
+import org.opensearch.ad.model.AnomalyResult;
+import org.opensearch.client.Client;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+public class ProfileUtil {
+    /**
+     * Create search request to check if we have at least 1 anomaly score larger than 0 after AD job enabled time
+     * @param detectorId detector id
+     * @param enabledTime the time when AD job is enabled in milliseconds
+     * @return the search request
+     */
+    private static SearchRequest createInittedEverRequest(String detectorId, long enabledTime, String resultIndex) {
+        BoolQueryBuilder filterQuery = new BoolQueryBuilder();
+        filterQuery.filter(QueryBuilders.termQuery(AnomalyResult.DETECTOR_ID_FIELD, detectorId));
```

Comment:
Historical analysis result also stored in result index, which has non-null `task_id`. For realtime detection result, we should filter `task_id == null`

Reply:
good callout. Changed.